Sources_v3/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -316,6 +316,85 @@ public extension _CurrentChatUserController {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        let deviceId = token.deviceToken
+        client
+            .apiClient
+            .request(
+                endpoint: .addDevice(
+                    userId: currentUserId,
+                    deviceId: deviceId
+                ),
+                completion: { [weak client] result in
+                    if let error = result.error {
+                        completion?(error)
+                        return
+                    }
+                    client?.databaseContainer.write({ (session) in
+                        try session.saveCurrentUserDevice(.init(id: deviceId))
+                    }) { completion?($0) }
+                }
+            )
+    }
+    
+    /// Removes a registered device from the current user.
+    /// `setUser` must be called before calling this.
+    /// - Parameters:
+    ///   - id: Device id to be removed. You can obtain registered devices via `currentUser.devices`.
+    ///   If `currentUser.devices` is not up-to-date, please make an `updateDevices` call.
+    ///   - completion: Called when device is successfully deregistered, or with error.
+    func removeDevice(id: String, completion: ((Error?) -> Void)? = nil) {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        client
+            .apiClient
+            .request(
+                endpoint: .removeDevice(
+                    userId: currentUserId,
+                    deviceId: id
+                ),
+                completion: { [weak client] result in
+                    if let error = result.error {
+                        completion?(error)
+                        return
+                    }
+                    client?.databaseContainer.write({ (session) in
+                        session.deleteDevice(id: id)
+                    }) { completion?($0) }
+                }
+            )
+    }
+    
+    /// Updates the registered devices for the current user from backend.
+    /// - Parameter completion: Called when request is successfully completed, or with error.
+    func updateDevices(completion: ((Error?) -> Void)? = nil) {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
+            return
+        }
+        client.apiClient.request(endpoint: .devices(userId: currentUserId)) { [weak self] (result) in
+            do {
+                let devicesPayload = try result.get()
+                self?.client.databaseContainer.write({ (session) in
+                    try session.saveCurrentUserDevices(payload: devicesPayload)
+                }) { completion?($0) }
+            } catch {
+                completion?(error)
+            }
+        }
+    }
```

Comment:
Don't we get this data when querying the user, aka when calling `synchronize`?

Reply:
We do, I've also added this API in case developers need to query devices separately. To be honest, I'm not so sure if we should have this in

Reply:
Since we don't have anything like this for other things, I don't see a reason why to have it here tbh.